src/evmc-client/src/client.rs

```diff
@@ -0,0 +1,381 @@
+use candid::utils::ArgumentEncoder;
+use candid::{CandidType, Principal};
+use did::block::BlockResult;
+use did::{
+    BasicAccount, BlockNumber, Bytes, Transaction, TransactionParams, TransactionReceipt, H160,
+    H256, U256,
+};
+use ic_exports::icrc_types::icrc1::account::Subaccount;
+use serde::Deserialize;
+
+use crate::{EvmResult, IcError, IcResult};
+
+#[async_trait::async_trait]
+pub trait EvmCanisterClient {
```

Comment:
This is not an `EvmCanisterClient` but a generic `CanisterClient`. We should think about moving it in the `canister-sdk` along with the agent and IC implementations

Reply:
Thinking about it, I think it would make more sense to have it in this repo, why do you think we should move it to the canister-sdk ?